src/plugin/systems.rs

```diff
@@ -131,6 +132,81 @@ pub fn apply_scale(
     }
 }
 
+/// System responsible for detecting changes in the hierarchy that would
+/// affect the collider's parent rigid body.
+pub fn collect_collider_hierarchy_changes(
+    mut commands: Commands,
+
+    mut hierarchy_events: EventReader<HierarchyEvent>,
+    parents: Query<&Parent>,
+    childrens: Query<&Children>,
+    rigid_bodies: Query<&RigidBody>,
+    colliders: Query<&Collider>,
+    mut collider_parents: Query<&mut ColliderParent>,
+) {
+    let child_colliders = |entity: Entity| -> Vec<Entity> {
+        let mut found = Vec::new();
+        let mut possibilities = vec![entity];
+        while let Some(entity) = possibilities.pop() {
+            if rigid_bodies.contains(entity) {
+                continue;
+            }
+
+            if colliders.contains(entity) {
+                found.push(entity);
+            }
+
+            if let Ok(children) = childrens.get(entity) {
+                possibilities.extend(children.iter());
+            } else {
+                continue;
+            };
+        }
+
+        found
+    };
```

Comment:
This could be done better, ideally without allocations. I'm thinking of just refactoring this all so that any entity in the hierarchy between a `RigidBody` and a `Collider` has a `ColliderLink` marker component.

Reply:
I suppose this is fine as long as the Vec is re-used in the system as a `Local<Vec<Entity>>`